### PR TITLE
Add space to the clipping region of physical model layers with shadows

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth

File: packages/flutter/lib/src/rendering/object.dart [hunk 1]
```diff
@@ -472,6 +472,11 @@ class PaintingContext {
       childContext._stopRecordingIfNeeded();
     } else {
       if (elevation != 0) {
+        // The drawShadow call doesn't add the region of the shadow to the
+        // picture's bounds, so we draw a hardcoded amount of extra space to
+        // account for the maximum potential area of the shadow.
+        canvas.drawRect(offsetBounds.inflate(20.0),
+                        new Paint()..color=const Color(0));
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/object.dart]: This seems pretty hacky (painting invisible black should be a no-op, inflating by 20 pixels seems arbitrary). Is there a long-term fix planned?

At a minimum we should add a TODO to fix this.

Comment (jason-simmons) [inline packages/flutter/lib/src/rendering/object.dart]: At some point Skia should provide a way to calculate the area drawn by the shadow and adjust the picture's size accordingly.  Until that's available, we need to do some other operation that causes the picture to extend its bounds.

Comment (Hixie) [inline packages/flutter/lib/src/rendering/object.dart]: Cool, in that case the TODO can just be "Remove this once skia does it for us". :-)

**Pull request merged**